### PR TITLE
skytemple: 1.3.2 -> 1.3.10

### PR DESCRIPTION
**Pull request by marius851000**
###### Motivation for this change
A new version was released, including an important (for me), as well as some other interesting feature.

###### Things done

As I needed to recompile basically everything, I didn't tested it with webgitgtk (disabled this input locally)

Also disabled a flaky test in a depency. It refused to build on the first try, but succeded in the second, with no change in library code.

Target staging due to using function stabilized in rust 1.59

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Ping @xfix

### CODE REVIEW

**Comment by xfix** [comment]
I think I would prefer to wait for Rust 1.59 to be in master to test this without having to build everything.

**Review by sugar700 — CHANGES_REQUESTED** [review]
Could we use system gettext in `skytemple-rust` by setting `GETTEXT_SYSTEM` environment variable during build? As expected by gettext-sys crate, see https://github.com/Koka/gettext-rs/tree/master/gettext-sys.

**Comment by marius851000** [comment]
Did that. And updated to SkyTemple 1.3.6 (it would indeed be a good idea to wait for rust 1.59 to hit master -- and the binary cache)

File: pkgs/development/python-modules/gbulb/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  propagatedBuildInputs = [
+    pygobject3
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "GLib implementation of PEP 3156";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gbulb/default.nix:27]: Missing homepage.

File: pkgs/development/python-modules/gbulb/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "gbulb";
+    rev = "v${version}";
+    sha256 = "sha256-QNpZf1zfe6r6MtmYMWSrXPsXm5iX36oMx4GnXiTYPaQ=";
+  };
+
+  propagatedBuildInputs = [
+    pygobject3
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gbulb/default.nix:24]: The package has tests and missing pythonImportsCheck.

**Comment by xfix** [comment]
Could this pull request be updated to use master as a target?

**Review by sugar700 — CHANGES_REQUESTED** [review]
Please rebase this pull request with master.

**Comment by marius851000** [comment]
Fixed the gbulb stuff, and updated to 1.3.8. Just an hour ago, 1.3.9 was released. I'll update this PR to use it.

**Comment by marius851000** [comment]
Should be good now.

File: pkgs/development/python-modules/py-desmume/default.nix [hunk 3]
```diff
@@ -3,33 +3,24 @@
...
-  postPatch = ''
-    cp -R --no-preserve=mode ${desmume} __build_desmume
-  '';
-
   buildInputs = [ GitPython libpcap SDL2 alsa-lib soundtouch openal ];
   nativeBuildInputs = [ meson ninja pkg-config ];
   propagatedBuildInputs = [ pillow pygobject3 ];
 
+  patchPhase = "ls .";
+
```
Comment (xfix) [inline pkgs/development/python-modules/py-desmume/default.nix:23]:
```suggestion
```

That line probably shouldn't have been here.

**Review by sugar700 — CHANGES_REQUESTED** [review]
Otherwise looks fine.

File: pkgs/development/python-modules/py-desmume/default.nix (hunk 3 above)
Comment (marius851000) [reply pkgs/development/python-modules/py-desmume/default.nix]: Thank. Fixed.

**Review by sugar700 — APPROVED** [review]

File: pkgs/development/python-modules/gbulb/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  disabledTests = [
+    "test_glib_events.TestBaseGLibEventLoop" # Somtimes fail due to imprecise timing
+  ];
+
+  pythonImportsCheck = [ "gbulb" ];
+
+  doCheck = true;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gbulb/default.nix:38]:
```suggestion
```
that is the default

**Review by sugar700 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 162931` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.pytest-sanic</li>
<li>python39Packages.spacy</li>
<li>python39Packages.spacy-transformers</li>
<li>python39Packages.textacy</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>dyndnsc: log was empty</li>
<li>gns3-gui: log was empty</li>
<li>gns3-server: log was empty</li>
<li>moz-phab: log was empty</li>
<li>netbox: log was empty</li>
<li>octoprint: log was empty</li>
<li>python39Packages.json-logging: log was empty</li>
<li>python39Packages.sanic-auth: <a href="https://termbin.com/6m3v">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/a8c97a1d85eecfe6a8aeb55348536d5f/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>

  <summary>18 packages failed to build and are new build failures:</summary>
  <ul>
<li>alerta-server: log was empty</li>
<li>cura: log was empty</li>
<li>home-assistant-component-tests.sentry: log was empty</li>
<li>moodle-dl: log was empty</li>
<li>openai: log was empty</li>
<li>(python39Packages.openai): log was empty</li>
<li>python310Packages.skytemple-dtef: log was empty</li>
<li>python310Packages.skytemple-files: log was empty</li>
<li>python310Packages.skytemple-ssb-debugger: log was empty</li>
<li>python39Packages.django-rq: log was empty</li>
<li>python39Packages.sentry-sdk: <a href="https://termbin.com/op6u">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/77793ccc0bbbe91517b8fee0d0ecf123/raw/index.html">pretty log </a></li>
<li>python39Packages.skytemple-dtef: log was empty</li>
<li>python39Packages.skytemple-files: log was empty</li>
<li>python39Packages.skytemple-ssb-debugger: log was empty</li>
<li>python39Packages.spacy-loggers: log was empty</li>
<li>python39Packages.wandb: log was empty</li>
<li>skytemple: log was empty</li>
<li>tribler: log was empty</li>
  </ul>
</details>
<details>
<summary>
13 packages built:
</summary>
<ul>
<li>python310Packages.dungeon-eos</li>
<li>python310Packages.gbulb</li>
<li>python310Packages.py-desmume</li>
<li>python310Packages.skytemple-rust</li>
<li>python39Packages.dungeon-eos</li>
<li>python39Packages.elastic-apm</li>
<li>python39Packages.entrance</li>
<li>python39Packages.entrance-with-router-features</li>
<li>python39Packages.gbulb</li>
<li>python39Packages.py-desmume</li>
<li>python39Packages.sanic</li>
<li>python39Packages.sanic-testing</li>
<li>python39Packages.skytemple-rust</li>
</ul>
</details>

**Comment by xfix** [comment]
Can this pull request be changed to target master, so that hopefully this change could end in 22.05?

**Comment by marius851000** [comment]
Done. Also put SkyTemple 1.3.10 that contain a minor bufgix.

File: pkgs/development/python-modules/sanic/default.nix [hunk 5]
```diff
@@ -102,6 +102,7 @@ buildPythonPackage rec {
     "test_check_timeouts_response_timeout"
     "test_reloader_live"
     "test_zero_downtime"
+    "test_noisy_exceptions"
```
Comment (xfix) [inline pkgs/development/python-modules/sanic/default.nix:105]:
```suggestion
```

Already dealt with by https://github.com/NixOS/nixpkgs/commit/2f7893fbf69006a303ab0b61d6e857a4e00b38d2, please rebase with newest master.

**Review by sugar700 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/skytemple/default.nix [hunk 6]
```diff
@@ -2,13 +2,13 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "skytemple";
-  version = "1.3.2";
+  version = "1.3.9";
```
Comment (xfix) [inline pkgs/applications/misc/skytemple/default.nix:5]:
```suggestion
  version = "1.3.10";
```

This should actually have 1.3.10 (also, update checksum, obviously).

**Review by sugar700 — APPROVED** [review]

**Pull request merged**